schema/filter_artworks.js

```diff
@@ -132,13 +133,22 @@ function filterArtworks(primaryKey) {
       },
     },
     resolve: (root, options, request, { rootValue: { accessToken } }) => {
+      const gravityOptions = options;
       if (primaryKey) {
-        options[primaryKey] = root.id; // eslint-disable-line no-param-reassign
+        gravityOptions[primaryKey] = root.id;
       }
+
+      // Support queries that have an empty, or * medium which indicates
+      // that we want all mediums. This results in us removing the query option
+      // making the graphQL queries between all and a subset of mediums the same shape.
+      if (options.medium === '' || options.medium === '*') {
+        delete gravityOptions.medium;
+      }
+
       if (accessToken) {
-        return gravity.with(accessToken)('filter/artworks', options);
+        return gravity.with(accessToken)('filter/artworks', gravityOptions);
       }
-      return gravity('filter/artworks', options);
+      return gravity('filter/artworks', gravityOptions);
```

Comment:
Can you check if this conditional is really needed or if it can work fine with `null` as `accessToken` ?